src/components/tasks/task-dialog.tsx

```diff
@@ -108,27 +118,73 @@ const TaskDialog = ({ projectId, milestoneId, open, task, onClose }: Props) => {
   });
 
   /**
-   * Upload task attachment mutation
+   * Upload task attachment handler function
+   * 
+   * @param file file to upload
    */
-  const uploadTaskAttachmentMutation = useMutation({
-    mutationFn: (file: File) => filesApi.uploadFile(file, TASK_ATTACHMENT_UPLOAD_PATH),
-    onSuccess: () => {
-      queryClient.invalidateQueries({ queryKey: ["taskAttachments"] });
+  const handleUploadNewAttachment = async (file: File) => {
+    try {
+      setFileUploadLoaderVisible(true);
+      const url = await filesApi.uploadFile(file, TASK_ATTACHMENT_UPLOAD_PATH);
+      const newAttachmentUrls = [...taskData.attachmentUrls, url];
+
+      setTaskData({ ...taskData, attachmentUrls: newAttachmentUrls });
+      setFileUploadLoaderVisible(false);
       setAttachmentDialogOpen(false);
-    },
-    onError: (error) => console.error(t("errorHandling.errorUploadingTaskAttachment"), error),
-  });
+    } catch (error) {
+      console.error(t("errorHandling.errorUploadingNewTaskAttachment"), error);
+    }
+  };
+
+  /**
+   * Handles existing attachment upload
+   * 
+   * @param attachmentUrl attachment url
+   */
+  const handleUploadExistingAttachment = async (attachmentUrl: string) => {
+    try {
+      setFileUploadLoaderVisible(true);
+      const newAttachmentUrlsIfDoesntExist = taskData.attachmentUrls.includes(attachmentUrl) ? taskData.attachmentUrls : taskData.attachmentUrls.concat(attachmentUrl);
+      setTaskData({ ...taskData, attachmentUrls: newAttachmentUrlsIfDoesntExist });
+      setAttachmentDialogOpen(false);
+      setFileUploadLoaderVisible(false);
```

Comment:
Maybe I've missed it but isn't this where we want to set some kind of notification to the user that the file uploaded already exists?

Reply:
Yes, I am adding it on my next PR for task connections